test/e2e/cypress/e2e/hosts_overview.cy.js

```diff
@@ -170,4 +170,41 @@ context('Hosts Overview', () => {
       });
     });
   });
+
+  describe('Deregistration', () => {
+    describe('Clean-up buttons should be visible only when needed', () => {
+      const heartbeatTargets = agents();
+      const failingHost = heartbeatTargets.shift();
+      before(() => {
+        cy.task('startAgentHeartbeat', heartbeatTargets);
+      });
+
+      it('should hide all Clean-up buttons when heartbeat starts except the first one', () => {
```

Comment:
I personally still don't like this testing approach.
I don't really think we should use so many agents. I would prefer to use one/two agent, and test thing more thorougly.
As commented, a thing that I would do is:
1. Send heartbeat for this agent. Check button is not there
2. Stop sending hearbeat. Check the button shows up. This is the most important test. I know that it takes time (10 seconds) to reach the heartbeat failed scenario, but otherwise we are not testing this explicitily
3. Send hearbeat again and check button goes away